mlrun/api/crud/marketplace.py

```diff
@@ -183,25 +194,56 @@ def get_source_catalog(
     def get_item(
         self,
         source: MarketplaceSource,
-        item_name,
-        channel,
-        version=None,
-        tag=None,
-        force_refresh=False,
+        item_name: str,
+        version: Optional[str] = None,
+        tag: Optional[str] = None,
+        force_refresh: bool = False,
     ) -> MarketplaceItem:
-        catalog = self.get_source_catalog(source, channel, version, tag, force_refresh)
-        items = [item for item in catalog.catalog if item.metadata.name == item_name]
-        if not items:
+        """
+        Retrieve item from source. The item is filtered by tag and version.
+
+        :param source:          Marketplace source object
+        :param item_name:       name of the item to retrieve
+        :param version:         version of the item
+        :param tag:             tag of the item
+        :param force_refresh:   if True, the catalog will be loaded from source always,
+                                otherwise will be pulled from db (if loaded before)
+
+        :return: marketplace item object
+
+        :raise if the number of collected items from catalog is not exactly one.
+        """
+        catalog = self.get_source_catalog(source, version, tag, force_refresh)
+        items, num_items = self._get_catalog_items_filtered_by_name(
+            catalog.catalog, item_name
+        )
+        if not num_items:
             raise mlrun.errors.MLRunNotFoundError(
-                f"Item not found. source={item_name}, channel={channel}, version={version}"
+                f"Item not found. source={item_name}, version={version}"
             )
-        if len(items) > 1:
+        if num_items > 1:
             raise mlrun.errors.MLRunInvalidArgumentError(
                 "Query resulted in more than 1 catalog items. "
-                + f"source={item_name}, channel={channel}, version={version}, tag={tag}"
+                + f"source={item_name}, version={version}, tag={tag}"
             )
         return items[0]
 
+    @staticmethod
+    def _get_catalog_items_filtered_by_name(
+        catalog: List[MarketplaceItem],
+        item_name: str,
+    ) -> Tuple[List[MarketplaceItem], int]:
+        """
+        Retrieve items from catalog filtered by name
+
+        :param catalog:     list of items
+        :param item_name:   item name to filter by
+
+        :return:   list of item objects from catalog
+        """
+        items = [item for item in catalog if item.metadata.name == item_name]
+        return items, len(items)
```

Comment:
```suggestion
        return items
```

I might have been over-explicit in my previous comment - there's really no point in returning `len(items)` as the caller can do it if it matters. I just meant to return the list with the items that match, regardless how many items are there

Reply:
Done